### PR TITLE
Migrate Kubernetes Schedules

### PR DESCRIPTION
**Pull request by cppwfs**
Migrates schedules that use SchedulerTaskLauncher and have the name in the  format `scheduleName-scdf-taskdefname` to a schedule in which the platform launches the schedule directly.

### CODE REVIEW

File: dataflow-migrate-schedules/README.adoc [hunk 1]
```diff
@@ -102,24 +101,75 @@ To enable this feature use the `runMigration.sh` script as follows.
...
+1) Establish the following environment variables:
+```
+export spring_profiles_active=kubernetes
+export spring_autoconfigure_exclude=org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryDeployerAutoConfiguration
+export spring_datasource_url=<The Spring Cloud Data Flow database URL to be used by the migration tool>
+export spring_datasource_username=<The Spring Cloud Data Flow database username to be used by the migration tool>
+export spring_datasource_password=<The Spring Cloud Data Flow database password to be used by the migration tool>
+export spring_datasource_driverClassName=<The Spring Cloud Data Flow database driverClassName to be used by the migration tool>
+export KUBERNETES_NAMESPACE=< The namespace for the scheduled apps >
+export dbDriverClassName=<The Spring Cloud Data Flow database driverClassName to be used by the migrated tasks>
```
Comment (chrisjs) [inline dataflow-migrate-schedules/README.adoc:127]: these seem duplicated given the exist with spring_ prefix above?

Comment (chrisjs) [reply dataflow-migrate-schedules/README.adoc:127]: understood why now re: dup -- the spring_datasource_url could be anything (ie: mysql port forwarded to localhost, etc) where in dbUrl you want the actual address of the DB

**Comment by chrisjs** [comment]
i installed SCDF 2.4.0, created a task, created a schedule, saw it run the tasks on schedule. then updated to SCDF 2.5.0.BUILD-SNAPSHOT image for SCDF, ran the migration tool and saw the task get migrated properly, a new cronjob object created, etc. tasks were running and being cleaned up as expected. deleting the schedule for that task appears to have worked correctly as well

one thing to note that may or may not be considered for this migration is the option to cleanup all the pods the TL left behind

so from a k8s standpoint, things seem to work as expected

@ilayaperumalg any additions or feedback on CF?

**Comment by cppwfs** [comment]
Updated the PR based @chrisjs comments as well as those we discussed today.  I will add the pod cleanup improvement in a later commit.

File: dataflow-migrate-schedules/pom.xml [hunk 2]
```diff
@@ -6,18 +6,18 @@
...
-	<name>Schedule Migrator</name>
-	<description>Migrates SCDF Schedules to the 2.3 format</description>
+	<name>Schedule CD Migrator</name>
+	<description>Migrates SCDF Schedules so that SchedulerTaskLauncher is no longer required.</description>
 
 	<properties>
 		<java.version>1.8</java.version>
-		<spring-cloud-data-flow.version>2.3.0.RC1</spring-cloud-data-flow.version>
+		<spring-cloud-data-flow.version>2.5.0.BUILD-SNAPSHOT</spring-cloud-data-flow.version>
 		<deployer.version>2.1.0.RELEASE</deployer.version>
```
Comment (ilayaperumalg) [inline dataflow-migrate-schedules/pom.xml:21]: The deployer.version needs to be changed to use the latest `2.3.0.BUILD-SNAPSHOT`.

File: dataflow-migrate-schedules/README.adoc (hunk 1 above)
Comment (ilayaperumalg) [reply dataflow-migrate-schedules/README.adoc:125]: After the migration, I see that the `spring_` properties are what would become the datasource properties for the scheduled tasks. Hence, using the `localhost` as the `spring_datasource_url` wouldn't work and this has to be the actual DB host address.

File: dataflow-migrate-schedules/README.adoc [hunk 3]
```diff
@@ -58,8 +58,6 @@ applications:
     spring_cloud_deployer_cloudfoundry_services: <your SCDF database service,your scheduler service>
     spring_cloud_scheduler_cloudfoundry_schedulerUrl: <URL to scheduler service>
     spring_profiles_active: cf
```
Comment (ilayaperumalg) [inline dataflow-migrate-schedules/README.adoc:60]: I wonder `cf` is used already from the existing migration sample. But, this should also include `cloudfoundry` as the MigrationService is activated based on the profile name `cloudfoundry`.

**Comment by ilayaperumalg** [comment]
Verified both the K8s and PCF using `2.3.0.RELEASE` as the base. 

  - Created simple and composed tasks
  - Ran the migration tool for all vs single schedule name
  - Verified the migration changes:
      K8s: The cronjobs now schedules the apps directly
      CF: the schedule launcher app got deleted and the PCF scheduler now schedules the apps directly.

File: dataflow-migrate-schedules/README.adoc (hunk 3 above)
Comment (cppwfs) [reply dataflow-migrate-schedules/README.adoc:60]: good catch.  I'll update.

**Comment by chrisjs** [comment]
merged in 48434d7e6140c48469e7f5aaaeca4ba48fc1b481

**Pull request closed without merging**